From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Fri, 24 Nov 2023 10:38:49 +0100
Subject: Makefile.groups: tlsa in packaging group ktls if
 KTLS_INCLUDE_TLSA=yes

- if not, then it is in separate group module_group_ktlsa

(cherry picked from commit a49c8d8d968e31a539e47db6c06a0756e4be55e3)
---
 src/Makefile.groups | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/Makefile.groups b/src/Makefile.groups
index ec0d8ac..7cb9586 100644
--- a/src/Makefile.groups
+++ b/src/Makefile.groups
@@ -88,12 +88,12 @@ mod_list_memcached=memcached
 # - modules depending on openssl library
 mod_list_tlsdeps=crypto tls
 
-# - modules depending on openssl (+curl) library
-mod_list_tlsdeps_curl=auth_identity
-
 # - modules depending on static openssl library
 mod_list_tlsa=tlsa
 
+# - modules depending on openssl (+curl) library
+mod_list_tlsdeps_curl=auth_identity
+
 # - modules depending on static wolfssl library
 mod_list_tls_wolfssl=tls_wolfssl
 
@@ -414,11 +414,13 @@ module_group_kmemcached=$(mod_list_memcached)
 # pkg tls module
 module_group_ktls_basic=$(mod_list_tlsdeps)
 
+ifeq ($(KTLS_INCLUDE_TLSA),yes)
 # pkg tls module with curl
-module_group_ktls=$(mod_list_tlsdeps) $(mod_list_tlsdeps_curl)
-
+module_group_ktls=$(mod_list_tlsdeps) $(mod_list_tlsdeps_curl) $(mod_list_tlsa)
+else
 # pkg tlsa module
 module_group_ktlsa=$(mod_list_tlsa)
+endif
 
 # pkg tls_wolfssl module
 module_group_ktls_wolfssl=$(mod_list_tls_wolfssl)
